File: autoit-mediapipe-com/src/binding/tasks/components/containers/keypoint.cc
#include "binding/tasks/components/containers/keypoint.h"

namespace mediapipe::tasks::components::containers {
	LocationData::RelativeKeypoint ConvertNormalizedKeypointToProto(NormalizedKeypoint* normalized_keypoint) {
		LocationData::RelativeKeypoint normalized_keypoint_proto;
		normalized_keypoint_proto.set_x(normalized_keypoint->x);
		normalized_keypoint_proto.set_y(normalized_keypoint->y);
		if (normalized_keypoint->label) {
			normalized_keypoint_proto.set_keypoint_label(*normalized_keypoint->label);
		}
		if (normalized_keypoint->score) {
			normalized_keypoint_proto.set_score(*normalized_keypoint->score);
		}
		return normalized_keypoint_proto;
	}
}

File: src/declarations/tasks/components/containers/keypoint.js
const ns = "mediapipe::tasks::components::containers";
const nsProgId = ns.replaceAll("::", ".");

module.exports = ({language}) => {
    const progid = `mediapipe.tasks.${ language }.components.containers.keypoint`;

    return [
        // expose keypoint properties like in mediapipe python
        [`${ progid }.`, "", ["/Properties"], [
            [`${ ns }::NormalizedKeypoint`, "NormalizedKeypoint", "", ["/R", "=this", "/S"]],
        ], "", ""],

        [`struct ${ nsProgId }.NormalizedKeypoint`, "", [`/progid=${ progid }.NormalizedKeypoint`, "/Simple", "/DC"], [
            ["float", "x", "", ["/RW"]],
            ["float", "y", "", ["/RW"]],
            ["std::optional<std::string>", "label", "", ["/RW"]],
            ["std::optional<float>", "score", "", ["/RW"]],
        ], "", ""],
    ];
};
